planetiler-core/src/main/java/com/onthegomap/planetiler/reader/GeoPackageReader.java

```diff
@@ -0,0 +1,108 @@
+package com.onthegomap.planetiler.reader;
+
+import com.onthegomap.planetiler.Profile;
+import com.onthegomap.planetiler.collection.FeatureGroup;
+import com.onthegomap.planetiler.config.PlanetilerConfig;
+import com.onthegomap.planetiler.stats.Stats;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Consumer;
+import mil.nga.geopackage.GeoPackage;
+import mil.nga.geopackage.GeoPackageManager;
+import mil.nga.geopackage.features.user.FeatureDao;
+import mil.nga.geopackage.geom.GeoPackageGeometryData;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.geometry.jts.WKBReader;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Geometry;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+
+/**
+ * Utility that reads {@link SourceFeature SourceFeatures} from the vector geometries contained in a GeoPackage file.
```

Comment:
I'd like to try migrating natural earth reader to this after we merge. That comes in a `.gpkg.zip` file. I'm wondering if we want this to have the same handling that [natural earth reader has](https://github.com/onthegomap/planetiler/blob/cb082a0566547934281891d4426ea5dbfd245f1c/planetiler-core/src/main/java/com/onthegomap/planetiler/reader/NaturalEarthReader.java#L89-L106) where it detects that and unzips it first if necessary or if that unzip step should be a reusable thing we could tack onto any reader...?  

The ideal situation is where the reader is OK reading directly out of the zip like shapefiles could do and we don't need the intermediate file. Other formats would likely be the same, but as far as I could tell sqlite needs to point at an actual file on disk.

Reply:
Ah! Misread your comment last time. Sure, I think that'd be useful to add here. Since sqlite files can't be read directly from a zip file I think we might need to change behavior in #423 if we want to support GPKG glob matching, but I'd like to get that PR merged first (or do this change as a follow up)

Reply:
Ok I merged #423, did you want to include the zip handling in this PR?

Reply:
How about doing it as a quick follow up? I can create a PR focused on adding natural earth support